Mono.Cecil/AssemblyReader.cs

```diff
@@ -166,6 +166,7 @@ protected override void ReadModule ()
 
 		public void ReadModule (ModuleDefinition module, bool resolve_attributes)
 		{
+			module.ReadingMode = ReadingMode.Immediate;
```

Comment:
@jbevain You didn't include this in the changes you sent to me.  I added this for 2 reasons.

1) It seemed like the easiest way to write a test that meant something.

2) I saw some other behaviors such as writing and symbol reading checked the reading mode.  It seemed like it would be better if the reading mode was updated.  However, I'm not certain this is correct.  Does this seem sensible to you?

Reply:
I say just move this to `ImmediateRead ()`

Reply:
Done